### ISSUE TITLE
lint: add append-combine checker

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#171]
Detects `append` chains to the same slice that can be done in a single `append` call.

**Before:**
```go
xs = append(xs, 1)
xs = append(xs, 2)
```
**After:**
```go
xs = append(xs, 1, 2)
```

There are both performance and style reasons for choosing suggested form.
Even if compiler will rewrite "before" info "after", second form usually is easier to read with less
code repetition (especially if slice name is long).

**Comment by Quasilyte** [comment]
Does trigger several times inside GOROOT:
```

$GOROOT/src/cmd/dist/test.go:605:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/dist/test.go:674:4: append-combine: found appends with 2 args
$GOROOT/src/cmd/cgo/util.go:48:3: append-combine: found appends with 3 args
$GOROOT/src/cmd/vet/asmdecl.go:425:4: append-combine: found appends with 2 args
$GOROOT/src/cmd/vet/asmdecl.go:430:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/vet/asmdecl.go:434:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/vet/asmdecl.go:438:3: append-combine: found appends with 3 args
$GOROOT/src/cmd/vet/asmdecl.go:443:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/vet/asmdecl.go:448:3: append-combine: found appends with 2 args
$GOROOT/src/crypto/tls/handshake_server_test.go:515:2: append-combine: found appends with 2 args
$GOROOT/src/cmd/link/internal/ld/dwarf.go:1397:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/link/internal/ld/lib.go:1208:2: append-combine: found appends with 2 args
$GOROOT/src/cmd/link/internal/ld/lib.go:1164:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/link/internal/ld/lib.go:1312:3: append-combine: found appends with 5 args
$GOROOT/src/crypto/x509/name_constraints_test.go:1728:2: append-combine: found appends with 2 args
$GOROOT/src/crypto/x509/name_constraints_test.go:1711:3: append-combine: found appends with 6 args
$GOROOT/src/strconv/quote.go:31:4: append-combine: found appends with 2 args
$GOROOT/src/strconv/quote.go:54:3: append-combine: found appends with 2 args
$GOROOT/src/strconv/quote.go:87:4: append-combine: found appends with 2 args
$GOROOT/src/net/http/socks_bundle.go:106:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/nm/nm_test.go:223:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/nm/nm_test.go:226:4: append-combine: found appends with 2 args
$GOROOT/src/cmd/nm/nm_test.go:229:4: append-combine: found appends with 2 args
$GOROOT/src/cmd/compile/internal/ssa/deadcode_test.go:146:4: append-combine: found appends with 2 args
$GOROOT/src/cmd/go/internal/envcmd/env.go:93:2: append-combine: found appends with 2 args
$GOROOT/src/archive/tar/writer_test.go:36:4: append-combine: found appends with 2 args
$GOROOT/src/cmd/compile/internal/gc/range.go:225:3: append-combine: found appends with 2 args
$GOROOT/src/cmd/compile/internal/gc/select.go:338:2: append-combine: found appends with 3 args
$GOROOT/src/cmd/compile/internal/gc/walk.go:3325:2: append-combine: found appends with 2 args
$GOROOT/src/cmd/compile/internal/gc/inl_test.go:152:3: append-combine: found appends with 3 args
$GOROOT/src/net/dnsclient.go:30:3: append-combine: found appends with 4 args
$GOROOT/src/net/mac.go:21:3: append-combine: found appends with 2 args
$GOROOT/src/net/interface_linux_test.go:19:2: append-combine: found appends with 2 args
$GOROOT/src/net/interface_linux_test.go:44:2: append-combine: found appends with 2 args
```
